### DISCUSSION TITLE
[General] Is there a self-hosted Faucet option for ZK Stack custom devnet?

### DISCUSSION THREAD

**Discussion by ghost** [zkSync-Community-Hub/zksync-developers#700] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I’m setting up a custom devnet using the ZK Stack and was wondering if there’s a way to deploy a self-hosted faucet for my environment. I need a faucet to distribute test tokens to developers on my devnet but couldn’t find any clear documentation for this use case.
Has anyone implemented a self-hosted faucet with the ZK Stack, or is there a recommended tool that integrates well with a custom ZK Rollup? Any guidance or resources would be greatly appreciated!

**Comment by alhenaai** [comment]
Currently, the ZKsync documentation does not provide specific details on setting up a self-hosted faucet for a custom ZK Stack devnet. However, you can utilize third-party faucets to access testnet funds, as mentioned in the ZKsync documentation. These third-party options might be adapted for a custom setup, depending on your specific requirements.

For more detailed guidance or to explore integration possibilities, you might consider reaching out to the ZKsync community or support for more tailored advice. Here's a link to the section on [Network Faucets](https://docs.zksync.io/ecosystem/network-faucets) which might offer some starting points.

If you have further questions or need specific assistance, feel free to reach out!

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by popzxc** [comment, marked as answer]
As far as I know, right now there is no open-source ZKsync-specific faucet.
However, ZKsync is L1-compatible, so any open-source faucet that works for L1, should work for ZKsync too.
I personally don't have experience using them, but, for example, [this one](https://github.com/chainflag/eth-faucet) seems to be maintained - you can try it.

**Comment by uF4No** [comment]
You can try to implement [this open source PoW faucet](https://github.com/pk910/PoWFaucet).
